### PR TITLE
Expose funs

### PR DESCRIPTION
**Pull request by rdaly525**
This adds the following functions:
inline_instance
module.params
context.new_namespace

### CODE REVIEW

**Comment by rdaly525** [comment]
Depends on https://github.com/rdaly525/coreir/pull/702

File: coreir/wireable.py [hunk 1]
```diff
@@ -71,6 +71,10 @@ def name(self):
         return libcoreir_c.COREInstanceGetInstname(self.ptr).decode()
 
 
+def inline_instance(instance):
+    assert isinstance(instance,Instance)
```
Comment (leonardt) [inline coreir/wireable.py]: Can we make this raise a `TypeError`? If you have time, a more general solution might be to create a simple decorator in a file `util.py` which we can use to type check functions like this, like `@typecheck(<expected_types>...)`.

Alternatively, I think we could annotate the type using Python 3 syntax, and use mypy to do it.

But a simple `if <cond>: raise TypeErrror` will work fine.

File: tests/test_inline.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        out = inst.select("out")
+    
+    def num_instances():
+        return len(top_def.instances)
+
+    assert num_instances() == 5
+    for i in range(5):
+        inst = top_def.instances[0]
+        coreir.inline_instance(inst)
+        assert num_instances() == 6+i
```
Comment (leonardt) [inline tests/test_inline.py]: Nice test!

File: coreir/wireable.py (hunk 1 above)
Comment (rdaly525) [reply coreir/wireable.py]: Just fixed

**Comment by leonardt** [comment]
Can you rerun the build after you make a new release of coreir with the new C-API methods? Should be good to merge/release after that.

**Comment by rdaly525** [comment]
@leonardt Passes!

**Pull request merged**